MODULE.bazel

```diff
@@ -0,0 +1,16 @@
+module(name = "boost", repo_name = "com_github_nelhage_rules_boost")
+
+bazel_dep(
+    name = "platforms",
+    version = "0.0.7",
+)
+
+boost_libs = use_extension("//:boost/boost.bzl", "boost_deps_extension")
+
+use_repo(boost_libs, "boost")
+use_repo(boost_libs, "bazel_skylib")
```

Comment:
bazel_skylib, zlib, bzip2, zstd, openssl are all availble in the official registry. How about using those versions? (Didn't look yet into the details how they are built TBH.)

@Attempt3035 Any plans finishing this PR?

Reply:
Hey @mvukov!
I've been doing a bunch of work over the last couple of months trying to get the entirety of boost into bzlmod. It's still a bit of a mess and work in progress, but made a PR so you can check it out here: https://github.com/bazelbuild/bazel-central-registry/pull/1280
Hopefully that will be the way forward! Will try and get it tidied up and some readme instructions so you can give it a go, most of the modules are working, just adding tests!

Reply:
Took a look at https://github.com/bazelbuild/bazel-central-registry/pull/1280. How do you maintain all those patches? (I really hope not by hand). Unless I am mistaken, for each boost module a separate archive is downloaded. What is the benefit compared to a single archive downloaded here?

Didn't dive into all the details, but this PR here and approach of this repo seem much more maintainable than https://github.com/bazelbuild/bazel-central-registry/pull/1280 TBH. If you go https://github.com/bazelbuild/bazel-central-registry/pull/1280 route, does that mean this repo becomes deprecated?

Any objection to finishing up this PR? seems almost ready.

Reply:
@mvukov  Off course, you need some scripts to generate all this stuff - we could place those scripts in this repo here. I also started for other libs to have some helper scripts to be able to switch to newer versions without all the boilerplate. Some people only want to use a single header library of boost - why would you then download 300 MB complete boost? I see many people only using boost.asio. Also in the long term, we could try to bring Bazel support directly to the boost repo itself. Then we could easily copy & paste our BUILD and MODULE since boost itself is organized in this "multirepo" approach. Smaller modules == less CI build traffic, less disk usage, etc. - there are some benefits